src/main/java/org/sagebionetworks/web/client/widget/doi/DoiWidgetV2ViewImpl.java

```diff
@@ -0,0 +1,106 @@
+package org.sagebionetworks.web.client.widget.doi;
+
+import org.gwtbootstrap3.client.ui.Icon;
+import org.gwtbootstrap3.client.ui.constants.IconType;
+import org.gwtbootstrap3.client.ui.html.Span;
+import org.sagebionetworks.web.client.DisplayUtils;
+import org.sagebionetworks.web.client.GlobalApplicationState;
+import org.sagebionetworks.web.client.security.AuthenticationController;
+
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.TextBox;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+public class DoiWidgetV2ViewImpl implements DoiWidgetV2View {
+	
+	GlobalApplicationState globalApplicationState;
+	AuthenticationController authenticationController;
+	@UiField
+	Icon clipboardIcon;
+	@UiField
+	TextBox doi;
+	@UiField
+	Span doiLabel;
+	@UiField
+	Span synAlertContainer;
+
+	Widget widget;
+	
+	public interface Binder extends UiBinder<Widget, DoiWidgetV2ViewImpl> {}
+	
+	@Inject
+	public DoiWidgetV2ViewImpl(GlobalApplicationState globalApplicationState,
+							   AuthenticationController authenticationController,
+							   Binder uiBinder) {
+		this.globalApplicationState = globalApplicationState;
+		this.authenticationController = authenticationController;
+		widget = uiBinder.createAndBindUi(this);
+
+		doi.addClickHandler(event -> copyDoiToClipboard());
```

Comment:
This is really cool.  Please refactor into reusable TextBoxWithClipboard (or better name) that wraps or extends TextBox.  
Should be able to use this widget directly in ui.xml file, see HelpWidget for example (needs empty constructor, ...).